database/query/profiles.sql

```diff
@@ -63,11 +63,40 @@ SELECT * FROM profiles WHERE id = $1 AND project_id = $2 FOR UPDATE;
 SELECT * FROM profiles WHERE lower(name) = lower(sqlc.arg(name)) AND project_id = $1 FOR UPDATE;
 
 -- name: ListProfilesByProjectID :many
-SELECT sqlc.embed(profiles), sqlc.embed(profiles_with_entity_profiles) FROM profiles JOIN profiles_with_entity_profiles ON profiles.id = profiles_with_entity_profiles.profid
+WITH helper AS(
+     SELECT pr.id as profid,
+            ARRAY_AGG(ROW(ps.id, ps.profile_id, ps.entity, ps.selector, ps.comment)::profile_selector) FILTER (WHERE ps.id IS NOT NULL) AS selectors
+       FROM profiles pr
+       LEFT JOIN profile_selectors ps
+         ON pr.id = ps.profile_id
+      WHERE pr.project_id = $1
+      GROUP BY pr.id
+)
+SELECT
+    sqlc.embed(profiles),
+    sqlc.embed(profiles_with_entity_profiles),
+    helper.selectors::profile_selector[] AS profiles_with_selectors
+FROM profiles
+JOIN profiles_with_entity_profiles ON profiles.id = profiles_with_entity_profiles.profid
+JOIN helper ON profiles.id = helper.profid
 WHERE profiles.project_id = $1;
```

Comment:
**suggestion (non-blocking):** I think you could get rid of the `FILTER` clause by moving the `LEFT JOIN` on the outer query. I expect it to work the same given the result is a slice.

```suggestion
WITH helper AS(
     SELECT pr.id as profid,
            ARRAY_AGG(ROW(ps.id, ps.profile_id, ps.entity, ps.selector, ps.comment)::profile_selector) AS selectors
       FROM profiles pr
       JOIN profile_selectors ps
         ON pr.id = ps.profile_id
      WHERE pr.project_id = $1
      GROUP BY pr.id
)
SELECT
    sqlc.embed(profiles),
    sqlc.embed(profiles_with_entity_profiles),
    helper.selectors::profile_selector[] AS profiles_with_selectors
FROM profiles
JOIN profiles_with_entity_profiles ON profiles.id = profiles_with_entity_profiles.profid
LEFT JOIN helper ON profiles.id = helper.profid
WHERE profiles.project_id = $1;
```

Reply:
suggestion taken!